defender/tests/Governance_Twitter_Bot.spec.js

```diff
@@ -0,0 +1,252 @@
+const secrets = {
+  governance_twitter_bot_governanceAddress: 'GOVERNANCE_ADDRESS',
+  governance_twitter_bot_appKey: 'TWITTER_APP_KEY',
+  governance_twitter_bot_appSecret: 'TWITTER_APP_SECRET',
+  governance_twitter_bot_accessToken: 'TWITTER_ACCESS_TOKEN',
+  governance_twitter_bot_accessSecret: 'TWITTER_ACCESS_SECRET',
+};
+const mockDecimals = 18;
+
+const mockContract = {
+  initialProposalId: jest.fn(),
+  proposalCount: jest.fn(),
+  state: jest.fn(),
+  proposals: jest.fn(),
+  comp: jest.fn(),
+  quorumVotes: jest.fn(),
+  decimals: jest.fn().mockReturnValue(mockDecimals),
+};
+
+// mock ethers
+jest.mock('ethers', () => ({
+  ...jest.requireActual('ethers'),
+  Contract: jest.fn().mockReturnValue(mockContract),
+}));
+
+const { ethers } = require('ethers');
+
+const exampleProposal = {
+  id: 1, // id
+  proposer: `0x${'a'.repeat(40)}`, // proposer 0xaaaa...
+  eta: 0, // eta
+  startBlock: 0, // startBlock
+  endBlock: 94784, // endBlock 1 day, 2 hours, 3 minutes, 4 seconds later
+  forVotes: ethers.BigNumber.from(0), // forVotes
+  againstVotes: ethers.BigNumber.from(0), // againstVotes
+  abstainVotes: ethers.BigNumber.from(0), // abstainVotes
+  canceled: false, // canceled
+  executed: false, // executed
+};
+
+// mock the defender-relay-client package
+const mockProvider = {
+  getBlock: jest.fn(),
+};
+const mockSigner = {
+  getAddress: jest.fn().mockReturnValue(ethers.constants.AddressZero),
+};
+
+jest.mock('defender-relay-client/lib/ethers', () => ({
+  DefenderRelayProvider: jest.fn().mockReturnValue(mockProvider),
+  DefenderRelaySigner: jest.fn().mockReturnValue(mockSigner),
+}));
+
+jest.mock('axios', () => jest.fn());
+const mockAxios = require('axios');
+
+mockAxios.get = jest.fn();
+
+jest.mock('axios-retry', () => jest.fn());
+
+// jest.mock('../governance_twitter_bot/autotask-1/twitter-api-v2', () => ({
+//   ...jest.requireActual('../governance_twitter_bot/autotask-1/twitter-api-v2'),
+//   TwitterApi: {
+//     // ...jest.requireActual('../governance_twitter_bot/autotask-1/twitter-api-v2'),
+//     v1: {
+//       tweet: jest.fn(),
+//       reply: jest.fn(),
+//     },
+//   },
+// }));
+
+const { TwitterApi } = require('../governance_twitter_bot/autotask-1/twitter-api-v2');
+
+// Spy on the tweet and reply methods. The first tweet will return with id_str: '1'
+const mockTweet = jest.spyOn(TwitterApi.prototype.v1, 'tweet').mockImplementation(() => ({ id_str: '1' }));
+// reply will prepend a 1 to the tweetID that it was passed. '1' => '11' => '111'
+const mockReply = jest.spyOn(TwitterApi.prototype.v1, 'reply').mockImplementation((msg, id) => ({ id_str: `1${id}` }));
```

Comment:
```suggestion
const mockReply = jest.spyOn(TwitterApi.prototype.v1, 'reply').mockImplementation((_msg, id) => ({ id_str: `1${id}` }));
```
Nit syntax suggestion to add underscore to unused variable

Reply:
Unused variable marked in commit 31c7668fa77b9fdc10ffa7eee1347ea88fe3baba